tutorial/practicals/LFRic/single_node/code/scaled_matrix_vector_kernel_mod.F90

```diff
@@ -0,0 +1,140 @@
+! BSD 3-Clause License
+!
+! Modifications Copyright (c) 2017-2020, Science and Technology Facilities Council
+!
+! Redistribution and use in source and binary forms, with or without
+! modification, are permitted provided that the following conditions are met:
+!
+! * Redistributions of source code must retain the above copyright notice, this
+!   list of conditions and the following disclaimer.
+!
+! * Redistributions in binary form must reproduce the above copyright notice,
+!   this list of conditions and the following disclaimer in the documentation
+!   and/or other materials provided with the distribution.
+!
+! * Neither the name of the copyright holder nor the names of its
+!   contributors may be used to endorse or promote products derived from
+!   this software without specific prior written permission.
+!
+! THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+! "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+! LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+! FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+! COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+! INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+! BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+! LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+! CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+! LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+! ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+! POSSIBILITY OF SUCH DAMAGE.
+! -----------------------------------------------------------------------------
+! Author R. W. Ford, STFC Daresbury Lab
+! Modified I. Kavcic, Met Office
+
+module scaled_matrix_vector_kernel_mod
+
+use argument_mod,            only : arg_type,                 &
+                                    GH_FIELD, GH_OPERATOR,    &
+                                    GH_READ, GH_INC,          &
+                                    ANY_SPACE_1, ANY_SPACE_2, &
+                                    CELL_COLUMN
+use fs_continuity_mod,       only : W3
+use constants_mod,           only : r_def, i_def
+use kernel_mod,              only : kernel_type
+
+implicit none
+
+private
+
+!-------------------------------------------------------------------------------
+! Public types
+!-------------------------------------------------------------------------------
+
+type, public, extends(kernel_type) :: scaled_matrix_vector_kernel_type
+  private
+  type(arg_type) :: meta_args(4) = (/                                  &
+       arg_type(GH_FIELD,    GH_INC,  ANY_SPACE_1),                    &
+       arg_type(GH_FIELD,    GH_READ, ANY_SPACE_2),                    &
+       ! modified so that the redundant computation example will run
+       !arg_type(GH_OPERATOR, GH_READ, ANY_SPACE_1, ANY_SPACE_2),      &
```

Comment:
So we should wipe the implementation of the kernel itself as it will no longer be correct?

Reply:
Oh my. I just copied the example without thinking about it so I've no idea how this would affect all of the tutorials - as the same code is used for all of them. Well we don't use redundant computation so it might be OK, and I'm not even sure what the issue is/was. I'm going to suggest we leave this for the moment and get everything on master. If I have time after slides etc I'll take a look (which I won't). Is that OK? It will probably affect all the halo exchanges etc.

Reply:
Yep, no worries.